### ISSUE TITLE
Xtensa: unable to compile code that needs clang headers

### ISSUE THREAD

**Issue by deadprogram** [tinygo-org/tinygo#3974]
This commit introduced a problem that only seems to appear when compiling for Xtensa:

https://github.com/tinygo-org/tinygo/commit/d801d0cd539447322228e9f1daa860774584b4d3

The TinyGo drivers has a test case that can reproduce it:

```
tinygo build -size short -o ./build/test.elf -target=m5stack-core2 ./examples/ft6336/basic/
# tinygo.org/x/drivers/delay
delay/sleep.go:10:10: fatal: 'stdbool.h' file not found
```

I have confirmed that this commit is the one that introduced the error.

The only reason this was not discovered sooner, is because of an error in the CI discovered by @trichner as part of https://github.com/tinygo-org/drivers/pull/614

**Comment by deadprogram** [comment]
cc @aykevl

**Comment by deadprogram** [comment]
Corrected by #3975

**Comment by deadprogram** [comment]
Closing since the issue was introduced in the current dev cycle. Thanks @aykevl for the help!
